docs/then_empty.md

```diff
@@ -0,0 +1,48 @@
+---
+title: \`then\` is an empty schema
+code: then_empty
+categories: correctness
+dialects: 2019-09, 2020-12, draft7
+autofixable: true
+---
+
+## Description
+`then` set to the empty schema (`{}`) does not restrict validation and is most likely ineffective.
+
+> **Message shown to user:**
+> Remove empty `then` or add restrictions.
+
+### Example 1
+<details><summary>Before</summary>
+
+```json
+{
+  "if": {
+    "properties": {
+      "flag": {
+        "const": true
+      }
+    }
+  },
+  "then": {}
+}
+```
+</details>
+
+<details><summary>After</summary>
+
+```json
+{
+  "if": {
+    "properties": {
+      "flag": {
+        "const": true
+      }
+    }
+  }
+}
+```
+</details>
```

Comment:
An `if` with no `then` or `else` has no effect. Should this autofix remove the `if` too?

Reply:
Yeah, I think it makes sense. As @Relequestual pointed out, it does mean some annotations might not be emitted, but I think relying on annotations coming from dangling `if`s makes no sense anyway?